### PR TITLE
removed options from generator

### PR DESCRIPTION
**Pull request by jbhatab**
This PR built to remove a few options from the installer and the express server.

### Code removed from installer
Heroku
Hot Module Reloader
Express Server
Bootstrap
Ruby and JS linting

If anyone's curious on my logic, I can explain why I removed each in more detail but overall I felt they were outside of the scope of the gem and created maintenance issues. 

### Documentation and example app
The documentation in the readme hasn't been updated to reflect and that will have to be a dialog on how we want to port that over. Most likely we just need to remove all the relevant documentation and create small 'how tos' for each of these topics, but we can just remove things for now and add the how tos later. In the same vein as this, I also think we need to decide how we are going to move forward with example applications. we have the spec/dummy app and the webpack tutorial. I think we should stick with a single place for these things so we can make maintenance more stable. 

### How the app is run in develop
I kept the Procfile to run the app and the npm run command in the root package json but I think we should remove both the root package json and the Procfile in favor of manually running both commands. People can setup their own dev enviroments. I think this is outside the scope of this PR though. The relevant messages that pop up when running the generator were changed to reflect this though.

This change is [Reviewable](https://reviewable.io/reviews/shakacode/react_on_rails/364)

### CODE REVIEW

**Comment by jbhatab** [comment]
@justin808 When we finalize this, I'll rebase and it and merge them all into one commit.

**Comment by justin808** [comment]
Looking good -- left you some comments.

---

Reviewed 42 of 42 files at r1.
Review status: all files reviewed at latest revision, 11 unresolved discussions.

---

*[CHANGELOG.md, line 9 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUgEZFSrfcJViEkHuB:-KEUgEZFSrfcJViEkHuC:-514994602) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/CHANGELOG.md#L9)):*
5.2.0

---

*[CHANGELOG.md, line 10 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUgIHZa-B7vA8ly2q7:-KEUgIHZa-B7vA8ly2q8:1628009591) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/CHANGELOG.md#L10)):*
removing does not sound like an enhancement

provide some color to the read

---

*[README.md, line 104 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUgUxHKYZ4wIPurBiT:-KEUgUxIeYOj-AKMSYFV:-1595456856) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/README.md#L104)):*
or directly on your layout

---

*[README.md, line 407 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUgdKSjBjbUOgECoP8:-KEUgdKSjBjbUOgECoP9:-548317242) ([raw file](https://github.com/shakacode/react_on_rails/blob/6107015bbcc659fb62efc04905d87030b6e3dc48/README.md#L407)):*
This is not part of the generator.

We can rebase your changes on top of the doc changes I'm doing. You'll need to manually merge in your README changes.

---

*[lib/generators/react_on_rails/templates/base/base/package.json.tt, line 5 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUhZn6FcitjGIPlNpI:-KEUhZn6FcitjGIPlNpJ:42886087) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/lib/generators/react_on_rails/templates/base/base/package.json.tt#L5)):*
This should be 5.10.0

---

*[lib/generators/react_on_rails/templates/base/base/client/package.json.tt, line 89 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUisakgoNqe6Kqohce:-KEUisakgoNqe6Kqohcf:897446609) ([raw file](https://github.com/shakacode/react_on_rails/blob/6107015bbcc659fb62efc04905d87030b6e3dc48/lib/generators/react_on_rails/templates/base/base/client/package.json.tt#L89)):*
Probably don't need this one:

https://github.com/gaearon/babel-plugin-react-transform

---

*[lib/generators/react_on_rails/templates/base/base/client/package.json.tt, line 113 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUinepyzBm093rXzOg:-KEUinepyzBm093rXzOh:1381471041) ([raw file](https://github.com/shakacode/react_on_rails/blob/6107015bbcc659fb62efc04905d87030b6e3dc48/lib/generators/react_on_rails/templates/base/base/client/package.json.tt#L113)):*
you don't need node-sass and the dev server

---

*[lib/generators/react_on_rails/templates/base/base/lib/tasks/assets.rake.tt, line 1 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUjRVrMBPRftN8KcWp:-KEUjRVrMBPRftN8KcWq:-19238935) ([raw file](https://github.com/shakacode/react_on_rails/blob/6107015bbcc659fb62efc04905d87030b6e3dc48/lib/generators/react_on_rails/templates/base/base/lib/tasks/assets.rake.tt#L1)):*
We refer to this file here: https://github.com/shakacode/react_on_rails/blob/master/docs%2Fadditional-reading%2Fheroku-deployment.md

So we should have this example somewhere before we remove it.

---

*[spec/dummy/Gemfile.lock, line 54 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUl0s9GhqdaHf_sNhw:-KEUl0s9GhqdaHf_sNhx:1372685045) ([raw file](https://github.com/shakacode/react_on_rails/blob/6107015bbcc659fb62efc04905d87030b6e3dc48/spec/dummy/Gemfile.lock#L54)):*
I'd suggest not making changes to /spec/dummy for this PR

---

*[spec/dummy/Gemfile.lock, line 260 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUlkvxQcrCSJrOlu9L:-KEUlkvxQcrCSJrOlu9M:978807131) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/spec/dummy/Gemfile.lock#L260)):*
why is this changing? this doesn't look right

beta 4 is released

---

*[spec/react_on_rails/support/.DS_Store, line 0 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUmQ41SWmuEHDA-kYp:-KEUmQ41SWmuEHDA-kYq:1539201696) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/spec/react_on_rails/support/.DS_Store#L0)):*
was this file already here? Did you modify this? This type of file should not be in the repo.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-:-KEUmx5p0XdSM-yiorBR:-2093757911)*

**Comment by jbhatab** [comment]
@justin808 anyway we can disable coveralls on this one since it will definitely reduce coverage from dropped options?

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-:-KEWWKL5YnlwxhoXa5-m:-759027792)*

**Comment by jbhatab** [comment]
*[README.md, line 407 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUgdKSjBjbUOgECoP8:-KEWX5hQaMfcvn1og_5s:-449334039) ([raw file](https://github.com/shakacode/react_on_rails/blob/6107015bbcc659fb62efc04905d87030b6e3dc48/README.md#L407)):*
@justin808 Is there a branch I can work off of. It would be good to work off of that instead of master.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364)*

**Comment by jbhatab** [comment]
*[CHANGELOG.md, line 10 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUgIHZa-B7vA8ly2q7:-KEWpvlfT9v0rinfwOAF:-1838013589) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/CHANGELOG.md#L10)):*
@justin808 how do I add color?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364)*

**Comment by justin808** [comment]
Review status: 36 of 42 files reviewed at latest revision, 11 unresolved discussions, some commit checks failed.

---

*[CHANGELOG.md, line 10 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUgIHZa-B7vA8ly2q7:-KEYbHmqa3g8dECWtGIl:-825074717) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/CHANGELOG.md#L10)):*
motivations, reasons, etc. Taking away stuff to be easier to support long term (when nothing is broken right now).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364)*

**Comment by justin808** [comment]
Getting there. Don't worry about coveralls. Let's make sure the assets.rake file is created by the base installer (not heroku specific), and let's make sure the tests are not removed for this file.

---

Reviewed 7 of 7 files at r2.
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[docs/additional-reading/heroku-deployment.md, line 68 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEZ1IzbMdzWs9e8j4C4:-KEZ1IzcO6SDH8bmxk8V:-1797372844) ([raw file](https://github.com/shakacode/react_on_rails/blob/43d69576086648a683f2986509c450129828f9a6/docs/additional-reading/heroku-deployment.md#L68)):*
Create a rake file to add webpack compilation to asset precompilation. You may already have this file if you used the React on Rails generator.

---

*[docs/additional-reading/heroku-deployment.md, line 70 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEZ1svHGtr7J9BGNTte:-KEZ1svIQrF8Dwesa__v:-456496192) ([raw file](https://github.com/shakacode/react_on_rails/blob/43d69576086648a683f2986509c450129828f9a6/docs/additional-reading/heroku-deployment.md#L70)):*
ruby?

---

*[spec/react_on_rails/support/.DS_Store, line  \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUmQ41SWmuEHDA-kYp:-KEZ28gYXLiC47FdTy1T:262477672) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/spec/react_on_rails/support/.DS_Store#L)):*
@jbhatab ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-:-KEZ2GAPZjitYeM6O2G9:-1716218365)*

**Comment by jbhatab** [comment]
*[CHANGELOG.md, line 10 \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUgIHZa-B7vA8ly2q7:-KEZNTYoDs4UKFCRVuCS:-709601772) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/CHANGELOG.md#L10)):*
I literally meant how do I add color. http://stackoverflow.com/questions/23904274/is-there-a-way-to-get-colored-text-in-github-flavored-markdown. It seems you can't add color anymore without using code blocks which would look weird. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364)*

**Comment by jbhatab** [comment]
*[spec/react_on_rails/support/.DS_Store, line  \[r1\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEUmQ41SWmuEHDA-kYp:-KEamSfta8GVq0Nrqwef:-2096001433) ([raw file](https://github.com/shakacode/react_on_rails/blob/76ffd016f42da73a91dae907cae13eefa294d514/spec/react_on_rails/support/.DS_Store#L)):*
I removed it.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364)*

**Comment by patrickgordon** [comment]
My thoughts are that this is a positive step - I think it simplifies the gem and separates concerns of what is gem and what can be done **with** the gem.

I'm happy to help with the documentation around how one might go about adding these things to their project:

- **Heroku**
- **Hot Module Reloader**
- Express Server
- **Bootstrap**
- Ruby and JS linting

The suspect that the ones in bold will be the most popular / frequently asked. It might even be worth writing some documents on how to use other CSS frameworks in a general sense. Popular ones like PureCSS or MaterializeCSS come to mind.

Happy to help in any way I can.

**Comment by justin808** [comment]
@jbhatab couple small comments. Get CI to pass and I'll merge this. 

I also saw you put back in asset.rake, but I couldn't find the test which ensures this file is created.

---

Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 15 unresolved discussions, some commit checks failed.

---

*[docs/additional-reading/heroku-deployment.md, line 68 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEcd0bmvr3hJtv1Bv8k:-KEcd0bmvr3hJtv1Bv8l:-2108830459) ([raw file](https://github.com/shakacode/react_on_rails/blob/f4e8480067b71a7184f299a101f654ec3fc625dc/docs/additional-reading/heroku-deployment.md#L68)):*
```
Ensure that you have an `assets.rake` file to ensure that your assets are built for pre-compilation. This file is created by the generator.
```

---

*[lib/generators/react_on_rails/base_generator.rb, line 127 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEceCRZWXCogYpoSQqw:-KEceCRZWXCogYpoSQqx:-1241542717) ([raw file](https://github.com/shakacode/react_on_rails/blob/f4e8480067b71a7184f299a101f654ec3fc625dc/lib/generators/react_on_rails/base_generator.rb#L127)):*
dots at end of line? blanks?

---

*[lib/generators/react_on_rails/templates/base/base/lib/tasks/assets.rake.tt, line 26 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEceHhfKolR22SryKJB:-KEceHhfKolR22SryKJC:914361720) ([raw file](https://github.com/shakacode/react_on_rails/blob/f4e8480067b71a7184f299a101f654ec3fc625dc/lib/generators/react_on_rails/templates/base/base/lib/tasks/assets.rake.tt#L26)):*
need a new line

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-:-KEcf-JJ2GnJHExkIwVl:-249609939)*

**Comment by jbhatab** [comment]
@justin808 It doesn't show up in the PR anymore because I just reverted it back. It's in there though.

@patrickgordon I think the simplification of this will help a ton so that way other people can start contributing in a more modular fashion like that. 

@justin808 I'm thinking we may want to do some more edits on the core readme. Also some of the additional readings stuff may be broken due to the fact they imply you ran the generator. It may be best to just leave them there and mark them in an 'old generator' state while we fix everything up. What do you think?

**Comment by jbhatab** [comment]
*[lib/generators/react_on_rails/base_generator.rb, line 127 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEceCRZWXCogYpoSQqw:-KEgIxL-0soK-Z-A_N2O:1675637715) ([raw file](https://github.com/shakacode/react_on_rails/blob/f4e8480067b71a7184f299a101f654ec3fc625dc/lib/generators/react_on_rails/base_generator.rb#L127)):*
removed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364)*

**Comment by jbhatab** [comment]
*[lib/generators/react_on_rails/templates/base/base/lib/tasks/assets.rake.tt, line 26 \[r3\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEceHhfKolR22SryKJB:-KEgJIf_-3CPoF9MX9xE:-499146009) ([raw file](https://github.com/shakacode/react_on_rails/blob/f4e8480067b71a7184f299a101f654ec3fc625dc/lib/generators/react_on_rails/templates/base/base/lib/tasks/assets.rake.tt#L26)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364)*

**Comment by justin808** [comment]
One little comment -- rake is not a language. probably should be ruby.

---

Reviewed 2 of 3 files at r4.
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---

*[docs/additional-reading/heroku-deployment.md, line 70 \[r2\]](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-KEZ1svHGtr7J9BGNTte:-KEiTgNEpeh47Lklh2LX:-521582871) ([raw file](https://github.com/shakacode/react_on_rails/blob/43d69576086648a683f2986509c450129828f9a6/docs/additional-reading/heroku-deployment.md#L70)):*
@jbhatab This should be ruby, I'm guessing.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-:-KEiU1VTySH9K1O6vCt8:-74063728)*

**Comment by justin808** [comment]
Reviewed 1 of 3 files at r4.
Review status: all files reviewed at latest revision, 12 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364)*

File: README.md [hunk 1]
```diff
@@ -404,52 +389,6 @@ The best source of docs is the main [ReactOnRails.js](node_package/src/ReactOnRa
   setOptions(options)
 ```
 
-## Hot Reloading View Helpers
```
Comment (justin808) [inline README.md:413]: @jbhatab this is not part of the generator change.

File: spec/dummy/Gemfile.lock [hunk 2]
```diff
@@ -322,4 +322,4 @@ DEPENDENCIES
   web-console
 
 BUNDLED WITH
-   1.11.2
+   1.11.2
```
Comment (justin808) [inline spec/dummy/Gemfile.lock:325]: @jbhatab Something is not right. Removing bootstrap-sass (OK ,not used) but the change to gemfile.lock is not right.

**Comment by justin808** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r5, 6 of 6 files at r6.
Review status: all files reviewed at latest revision, 13 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/shakacode/react_on_rails/364#-:-KEs4pnunsUilTJyI2Qu:1417088721)*

**Pull request merged**